app.rb

```diff
@@ -0,0 +1,125 @@
+# App Console Entry Point
+require_relative 'classes/book'
+require_relative 'classes/person'
+require_relative 'classes/teacher'
+require_relative 'classes/student'
+require_relative 'classes/rental'
+require_relative 'classes/classroom'
+
+class App
+  attr_accessor :books, :people, :rentals
+
+  def initialize
+    @books = []
+    @people = []
+    @rentals = []
+  end
+
+  def books()
+    puts 'Listing all books:'
+    puts ' '
+    @books.each do |book|
+      puts "Title: \"#{book.title}\", Author: #{book.author}"
+    end
+    puts ' '
+  end
+
+  def people()
+    puts 'Listing all people:'
+    puts ' '
+    @people.each do |person|
+      puts "[#{person.class}] - Name: #{person.name}, ID: #{person.id}, Age: #{person.age}"
+    end
+    puts ' '
+  end
+
+  def add_book()
+    puts 'Creating a book:'
+    puts ' '
+    print 'Title: '
+    title = gets.chomp
+    print 'Author: '
+    author = gets.chomp
+    book = Book.new(title, author)
+    @books.push(book)
+    puts 'Book created successfully'
+    puts ' '
+  end
+
+  def add_student()
+    puts 'Creating a student:'
+    puts ' '
+    print 'Age: '
+    age = gets.chomp.to_i
+    print 'Name: '
+    name = gets.chomp
+    print 'Has parent permission? [Y/N]: '
+    permission = gets.chomp.upcase
+    permission = (permission == 'Y')
+    puts 'testing before student creation'
+    student = Student.new(name, age, permission)
+    puts 'testing after student creation'
+    @people.push(student)
+    puts 'Student created successfully'
+    puts ' '
+  end
```

Comment:
- [ ] Great job working on the functionality so far but when I try to create a new student, I face this error with the wrong number of arguments (given 3, expected 1..2) so kindly check on this error and try to solve the problem to let the user be able to create a student ⭐🔥💯🙏🏻

- When I try to create a student  ⬇
![image](https://github.com/Abdulwhbhussain/OOP-school-library/assets/58553711/94afb70f-59d4-4f07-a46e-e37edc4c7d14)

Reply:
Solved Bro!